cmd/main.go

```diff
@@ -44,18 +40,35 @@ coap-cli post channels/0bb5ba61-a66e-4972-bab6-26f19962678f/messages/subtopic -a
 `
 )
 
+var (
+	errCreateClient     = errors.New("failed to create client")
+	errSendMessage      = errors.New("failed to send message")
+	errInvalidObsOpt    = errors.New("invalid observe option")
+	errFailedObserve    = errors.New("failed to observe resource")
+	errTerminatedObs    = errors.New("observation terminated")
+	errCancelObs        = errors.New("failed to cancel observation")
+	errCodeNotSupported = errors.New("message can be GET, POST, PUT or DELETE")
+)
+
+type request struct {
+	code codes.Code
+	path string
+	host *string
+	port *string
+	cf   *int
+	data *string
+	auth *string
+	obs  *bool
+}
+
 func parseCode(code string) (codes.Code, error) {
-	switch code {
-	case get:
-		return codes.GET, nil
-	case put:
-		return codes.PUT, nil
-	case post:
-		return codes.POST, nil
-	case delete:
-		return codes.DELETE, nil
+	ret, err := codes.ToCode(code)
```

Comment:
Let's fix this in case an error happens during parsing (i.e. not all errors are unsupported code errors):
```go
func parseCode(code string) (codes.Code, error) {
	ret, err := codes.ToCode(code)
	if err != nil {
		return 0, err
	}
	switch ret {
	case codes.GET, codes.PUT, codes.POST, codes.DELETE:
		return ret, nil
	default:
		return 0, errCodeNotSupported
	}
}
```